src/bokeh/io/export.py

```diff
@@ -261,8 +258,8 @@ def get_screenshot_as_png(obj: UIElement | Document, *, driver: WebDriver | None
     with _tmp_html() as tmp:
         theme = (state or curstate()).document.theme
         html = get_layout_html(obj, resources=resources, width=width, height=height, theme=theme)
-        with open(tmp.path, mode="w", encoding="utf-8") as file:
-            file.write(html)
+        tmp.write(html.encode("utf-8"))
+        tmp.close()
```

Comment:
```suggestion
        with tmp as f:
            f.write(html.encode("utf-8"))
```
Should also work:

<img width="677" height="264" alt="image" src="https://github.com/user-attachments/assets/4f7fb2d0-0539-486a-a7d7-5b34eec2f76a" />

Reply:
[37e66e8](https://github.com/bokeh/bokeh/pull/14640/commits/37e66e8ceab404b8e38b1cd6eac9cacb4022ed9c) nested context managers always put me off for some reason but I guess it's ok